### ISSUE TITLE
Not disabling on iframes of sites that match disabled rules

### ISSUE THREAD

**Issue by yang** [gdh1995/vimium-c#709]
I added a site to the disabled rules, but the extension is still running on iframes. Is there a way to prevent this from happening? Thanks for the great extension.

**Comment by gdh1995** [comment]
It's designed to do so. Therefore, you need to add exclusion rules for those 3rd-party iframes, too.

You may click an iframe's content, and click Vimium C icon, and then Vimium C's popup dialog will auto predict an exclusion rule whose `pattern` is the iframe URL.

**Comment by yang** [comment]
If you are open to adding an option for this, I can look into contributing it

**Comment by yang** [comment]
(Fyi, my use case is a web application that has many different origins as iframe content. It is tedious to add every individual sites to the exception list.)

**Comment by gdh1995** [comment]
Then I wonder why all frames of those origins are not suitable to run Vimium C commands. In theory, pages from different (top-level) origins can not visit each other's JS world, so they should not have any consistent shortcut module?

If those origins belong to a same top-level origin, then exclusion patterns support RegExp and should be able to match them.

**Comment by yang** [comment]
Yeah, they don't belong to the same TLD. They communicates cooperatively using for instance postMessage. They interact poorly with Vimium because the outer application is a complex web page editor, and Vimium interferes with various keyboard operations.
